tiledb/ml/readers/_tensor_schema.py

```diff
@@ -221,11 +235,15 @@ def max_partition_weight(self) -> int:
         rows_per_slice = dim_tiles[0]
 
         # Reading a slice of `rows_per_slice` rows requires reading a number of tiles that
-        # depends on the size and tile extent of each dimension after the first one.
+        # depends on the size, secondary slice, and tile extent of each dimension after the first one.
         assert len(self.shape) == len(dim_tiles)
-        tiles_per_slice = np.prod(
-            [ceil(size / tile) for size, tile in zip(self.shape[1:], dim_tiles[1:])]
-        )
+        
+        tiles_per_slice = 1
+        for i, tile in enumerate(dim_tiles[1:], 1):
+            if i in self._secondary_slices and isinstance(self._secondary_slices[i], list):
+                tiles_per_slice *= len(set([(x - self._ned[i][0]) // tile for x in self._secondary_slices[i]]))
```

Comment:
1. Extract ` self._ned[i][0])` from this line to a variable
2. Can you elaborate how did you come up with this expression? It's not clear how this calculates the correct tiles per slice.

Reply:
Since tiles get stored starting from the beginning of the DenseArray, `x - self._ned[i][0]) // tile` gives the tile that the index `i` will be stored on. We then count the number of unique tiles to get the total number of tiles that will be downloaded on that dimension.

Reply:
Got it, thanks :+1: Please add a comment to that effect above this line.